python3/Makefile

```diff
@@ -20,3 +20,5 @@ install:
 	$(IPROG) bin/hfx_filename $(DESTDIR)$(OPTDIR)/bin
 	$(IPROG) bin/perfmon $(DESTDIR)$(OPTDIR)/bin
 	$(IPROG) bin/xe-scsi-dev-map $(DESTDIR)$(OPTDIR)/bin
+	mkdir -p $(DESTDIR)$(PLUGINDIR)
```

Comment:
From a reliable packaging perspective, using `mkdir -p something` raises an eyebrow with me:

The issue for me is that this `mkdir` does not specify the mode (permissions) of the directory.

While it's not an issue per se because unless something changes the default `umask` from `0022', the `mkdir` will create the directory with the mode `0755`, which is what we expect, this mkdir does not specify this to always be the case. It is left for the reviewer to guess that the `umask` is indeed what we expect.

The standard practise (at least for installation when packaging RPMs at distros based on Fedora and openSUSE) is to instead use `install -d -m 755` which is very clear during review as it specifies the mode of the directory and otherwise is the same as `mkdir -p`:

```suggestion
	install -d -m 755 $(DESTDIR)$(PLUGINDIR)
```
I'd prefer to ask @ashwin9390 to replace those `mkdir -p` to such `install -d -m 755` just for clarity during review when we move the files.

Agree?

I know this is really a nitpick, doing it this way is established and I think we should follow the best practises as we are now moving the code and if we don't start now, we never start to follow such practises.

Of course, for this to be effective, all mkdir's have to be replaced, but if we don't start, we don't get there. While not required to start with this PR, this line is one where we can start.

Also, to not clutter the new python3/Makefile with different `mkdir -p` or `install -d -m 755` all over the place, I'd prefer if we'd make it a point to move all directory creation to the top of the makefile's install rule where all other directories are created to avoid having to search for outliers all over the place.

While maybe not a realistic scenario, e.g. some sourced install script could set a different `umask` and with `mkdir -p` the mode of those directories would then be changed. Again, this is highly unlikely, but so we thought for the `xz-utils` attack as well.